zips/zip-0233.md

```diff
@@ -19,224 +20,146 @@ The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED", "OPTIONAL",
-$\mathsf{ZsfBalanceAfter}(\mathsf{height})$ as part of a new block commitment
-in the block header, at the end of the `hashBlockCommitments` chain in
-[ZIP-244](https://zips.z.cash/zip-0244#block-header-changes).
+## Transaction Format
 
-TODO ZIP editors: consider introducing a chain state commitment hash tree.
-(When we get closer to the network upgrade, we will have a better idea what
-commitments that network upgrade needs.)
+The following field is added to the v7 transaction format [3]:
 
-## Deposits into the Sustainability Fund
++-------+----------------+------------+------------------------------------------------------+
+| Bytes | Name           | Data Type  | Description                                          |
++=======+================+============+======================================================+
+|   8   | ``burnAmount`` | ``uint64`` | The amount of input value to be burned, in zatoshis. |
++-------+----------------+------------+------------------------------------------------------+
 
-Sustainability fund deposits can be made via the new `zsfDeposit` field. This
-can be done by:
-- ZEC fund holders, in non-coinbase transactions;
-- Zcash miners, in coinbase transactions.
+**Note:** Older transaction versions can continue to be supported after a
+network upgrade, but burning is not possible for these transactions. For
+example, NU5 supports both v4 and v5 transaction formats, for both coinbase and
+non-coinbase transactions.
 
-In transaction versions without this new field:
-- unclaimed miner fees and rewards in **coinbase transactions** are re-defined
-  as deposits into the sustainability fund, and
-- there are no sustainability fund deposits from non-coinbase transactions.
+## Consensus Rules
 
-Note: older transaction versions can continue to be supported after a network
-upgrade. For example, NU5 supports both v4 and v5 transaction formats, for both
-coinbase and non-coinbase transactions.
+The burned value must now be considered when calculating the total output value
+of a transaction. It should be treated similarly to a transparent output,
+except that there is no way for this value to be used as an input in a future
+transaction.
 
-### `zsfDeposit` fields in transactions
+## Digests
 
-Each transaction can dedicate some of its excess funds to the ZSF, and the
-remainder becomes the miner fee, with any excess miner fee/reward going to the
-ZSF
+The transaction digest algorithm defined in ZIP 244 [4] is to be modified for v7
+transactions as follows:
 
-This is achieved by adding a new field to the common transaction fields
-[#zip-0225-transaction-format]:
+A section is inserted, specified to read:
 
-- `zsfDeposit : u64 [zatoshi]`
+> Section T.5: `burn_amount_digest`
+>
+> A BLAKE2b-256 hash of the following values:
+>
+> ```
+> T.5a: burnAmount (8-byte little-endian value)
+> ```
+>
+> The personalization field of this hash is set to:
+>
+> ```
+> ZTxBurnAmnt_Hash
+> ```
```

Comment:
This will instead be included in the header digest; it does not require a separate section. If there is a specific rationale for including this as an additional section instead, please state that rationale.

Reply:
This sounds OK to me. I'll adjust the ZIP and the implementation PRs :+1:

Reply:
Fixed with this diff: https://github.com/zcash/zips/compare/d590947513d9c8cb2c5e772ecacf56c8b3cd1246..7e0431dc58e6236d9b7b051dc020435075780899